src/python/grpcio_observability/grpc_observability/_open_telemetry_observability.py

```diff
@@ -80,26 +112,39 @@ def __init__(
 
         self.exporter = _OpenTelemetryExporterDelegator(_plugins)
 
+    def __enter__(self):
+        global _OPEN_TELEMETRY_OBSERVABILITY  # pylint: disable=global-statement
+        with _observability_lock:
+            if _OPEN_TELEMETRY_OBSERVABILITY:
+                raise RuntimeError(
+                    "gPRC Python observability was already initiated!"
+                )
+            self.observability_init()
+            _OPEN_TELEMETRY_OBSERVABILITY = self
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb) -> None:
```

Comment:
Sam here. Can we share an implementation with `end_open_telemetry_observability`?